### ISSUE TITLE
host_vars do not get fully loaded when using 'inventory' directory and --limit

### ISSUE THREAD

**Issue by xytis** [ansible/ansible#13557]
##### Issue Type:
- Bug Report
- Bugfix Pull Request
##### Ansible Version:
- 1.9.4
##### Ansible Configuration:

```
[defaults]
inventory=inventory
```
##### Environment:

```
./inventory/g
./host_vars/h1
./host_vars/h2
./playbook.yml
```
##### Summary:

host_vars are not loaded when using inventory directory and --limit flag
##### Steps To Reproduce:

```
#./inventory/g
h1
h2
```

```
#./host_vars/h1

---
key: h1
```

```
#./host_vars/h2

---
key: h2
```

```
#./playbook.yml

---

- name: test host_vars includes
  gather_facts: no
  hosts: all
  tasks:
    - debug: var=hostvars['h1'].key
    - debug: var=hostvars['h2'].key
```

```
ansible-playbook playbook.yml -l h1
```
##### Expected Results:

```
PLAY [test host_vars includes] ************************************************

TASK: [debug var=hostvars['h1'].key] ******************************************
ok: [h1] => {
    "var": {
        "hostvars['h1'].key": "h1"
    }
}

TASK: [debug var=hostvars['h2'].key] ******************************************
ok: [h1] => {
    "var": {
        "hostvars['h2'].key": "h2"
    }
}

PLAY RECAP ********************************************************************
h1                         : ok=2    changed=0    unreachable=0    failed=0
```
##### Actual Results:

```
PLAY [test host_vars includes] ************************************************

TASK: [debug var=hostvars['h1'].key] ******************************************
ok: [h1] => {
    "var": {
        "hostvars['h1'].key": "h1"
    }
}

TASK: [debug var=hostvars['h2'].key] ******************************************
ok: [h1] => {
    "var": {
        "hostvars['h2'].key": "{{hostvars['h2'].key}}"
    }
}

PLAY RECAP ********************************************************************
h1                         : ok=2    changed=0    unreachable=0    failed=0
```

**Comment by chrrrles** [comment]
I can duplicate this only on stable-1.9, not devel or stable-2.0.

**Comment by feliksik** [comment]
Duplicate of #12174. 

Hostvars are looked up relative to inventory path. 
If you move the inventory file to the same dir as the playbook,  it works. 

Ansible 2 is not affected.

**Comment by feliksik** [comment]
Similar to #12174, but you seem to not explicitly specify the inventory file. 

Hostvars are looked up relative to inventory path. 
If you move the inventory file to the same dir as the playbook,  it works. Specify with -i. 

Ansible 2 is not affected.

**Comment by abadger** [comment]
This shoukd be fixed by #15130 so going ahead and closing this.
